app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,34 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance_index = self._get_first_group_instance_index_in_block(group, blocks)
+            all_blocks_skipped = first_group_instance_index is None
 
-            if blocks:
-                path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
+            if not all_blocks_skipped:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance_index, first_block_in_group)
```

Comment:
Is it possible that `first_block_in_group` might be skipped, does it matter?

Reply:
It was an issue , but I changed the setting of it so that it is reset on every iteration of group. Blocks are mandatory within a group so it will be valid on each iteration.

Reply:
It does not matter, but we did not have a test to proove that . Modified an existing test to cover that scenario.